File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
from collections import deque

def solution(k, m, score):
    answer_lst = []
    score_queue = deque(sorted(score, reverse=True))
    
    while len(score_queue) >= m:
        cur_lst = []
        for i in range(m): 
            cur_lst.append(score_queue.popleft())
        answer_lst.append(min(cur_lst) * m)
    
    return sum(answer_lst)
    
        

File: 프로그래머스/1/135808. 과일 장수/과일 장수.js
const solution = (k, m, score) => {
    const score_queue = score.sort((a, b) => b - a);
    const result = [];
    let pointer = 0;
    
    while (pointer <= score_queue.length-m) {
        const cur_store_lst = [];
        for (let i=0; i<m; i++) {
            cur_store_lst.push(score_queue[pointer]);
            pointer ++;
        }
        result.push(Math.min(...cur_store_lst) * m);
    }
    return result.reduce((acc, cur) => acc + cur, 0);
}